src/test/java/dev/aherscu/instancio/experiment/PersonRequestHandlerTest.java

```diff
@@ -1,58 +1,77 @@
 package dev.aherscu.instancio.experiment;
 
-import static org.hamcrest.MatcherAssert.*;
-import static org.hamcrest.Matchers.*;
+import dev.aherscu.instancio.experiment.logic.PersonRequestHandler;
+import dev.aherscu.instancio.experiment.model.Address;
+import dev.aherscu.instancio.experiment.model.CreatePersonRequest;
+import dev.aherscu.instancio.experiment.model.CreatePersonResponse;
+import dev.aherscu.instancio.experiment.model.Person;
+import dev.aherscu.instancio.experiment.model.Phone;
+import dev.aherscu.instancio.experiment.utils.Tuple;
+import org.instancio.Gen;
+import org.instancio.Instancio;
+import org.instancio.Model;
+import org.instancio.TargetSelector;
+import org.instancio.generator.Generator;
+import org.instancio.generator.GeneratorSpec;
+import org.instancio.generator.specs.StringSpec;
+import org.instancio.settings.Keys;
+import org.instancio.settings.Settings;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
 
-import dev.aherscu.instancio.experiment.logic.*;
-import dev.aherscu.instancio.experiment.model.*;
-import org.testng.annotations.*;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.both;
+import static org.hamcrest.Matchers.empty;
+import static org.hamcrest.Matchers.hasProperty;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
+import static org.instancio.Select.all;
+import static org.instancio.Select.field;
 
 public class PersonRequestHandlerTest {
 
+    private static Model<CreatePersonRequest> createValidRequestModel() {
+        return Instancio.of(CreatePersonRequest.class)
+                .withSettings(Settings.create()
+                        .set(Keys.BEAN_VALIDATION_ENABLED, true))
+                .toModel();
+    }
+
     @DataProvider
     private static CreatePersonRequest[] data() {
-        return new CreatePersonRequest[] {
-            CreatePersonRequest.builder()
-                .person(Person.builder()
-                    .address(Address.builder()
-                        .line1("1")
-                        .city("A")
-                        .country("A")
-                        .zipCode("1")
-                        .build())
-                    .phone(Phone.builder()
-                        .countryCode(1)
-                        .number(1)
-                        .build())
-                    .build())
-                .build(),
-        };
+        return Instancio.stream(createValidRequestModel())
+                .limit(100)
+                .toArray(CreatePersonRequest[]::new);
```

Comment:
There is no guarantee here that `Phone#countryCode` will be generated with value of 1, for example.

Reply:
That's right. It will generate a valid number, which is in the [1..999] range. The higher the sample size, the more permutations will be tested. But to test boundary conditions only, then the value would need to be specified manually (or maybe via some custom logic; I'm not sure how that might look like).